react/features/base/tracks/actions.js

```diff
@@ -397,19 +397,29 @@ export function trackAdded(track) {
     return async (dispatch, getState) => {
         track.on(
             JitsiTrackEvents.TRACK_MUTE_CHANGED,
-        () => {
-            if (getSourceNameSignalingFeatureFlag(getState()) && track.getVideoType() === VIDEO_TYPE.DESKTOP) {
-                dispatch(screenshareTrackMutedChanged(track));
+            () => {
+                dispatch(trackMutedChanged(track));
+
+                if (getMultipleVideoSupportFeatureFlag(getState()) && track.getVideoType() === VIDEO_TYPE.DESKTOP) {
+                    dispatch(screenshareTrackMutedChanged(track));
```

Comment:
Can we add multi-stream handling in /base/tracks/middleware instead of dispatching another action here ?

Reply:
yeah I can move that into the middleware instead of dispatching here. What are your general thoughts on dispatching actions in middlewares? For our use case, will it be better to streamline the actions (`TRACK_ADDED` and `TRACK_UPDATED`) that triggers a fake SS instead of being more explicit on when to create one?